lighthouse-core/gather/driver.js

```diff
@@ -286,18 +286,19 @@ class Driver {
       }
     }
     return new Promise(async (resolve, reject) => {
-      const asyncTimeout = setTimeout((_ => {
+      const asyncTimeout = timeout > 0 ? setTimeout((_ => {
```

Comment:
I would have gone that route first, but it would need to be `let`, and my personal preferences led me to choose the ternary before reaching for `let`. wdyt

Reply:
whoops, yeah, `let`. And what's with that explicit `undefined`. Going to edit my comment because my brain was apparently off while writing that :)

The main issue is that you have to look down four lines to see the other half of the conditional, and the syntax makes it harder to parse than normal control flow mechanisms (like an if block). Generally long ternaries just aren't worth it, but at the very least, the null branch should be first